enclave_manager/avalon_enclave_manager/base_enclave_manager.py

```diff
@@ -132,21 +124,30 @@ def _setup_enclave(self):
         functionalities over this instance of the enclave. Therefore
         all relevant data is sent back here. This is kept in memory
         for use hereafter.
-
-        Returns :
-            signup_data - Relevant signup data to be used for requests to the
-                          enclave
         """
         try:
             logger.info("Initialize enclave and create signup data")
             signup_data = self._create_signup_data()
             if signup_data is None:
                 logger.error("Failed to create signup data")
-                return None
+                raise Exception("Failed to create signup data")
         except Exception as e:
-            logger.exception("failed to initialize/signup enclave; %s", str(e))
+            logger.exception("Failed to initialize/signup enclave; %s", str(e))
             sys.exit(-1)
-        return self._get_JSON_from_signup_object(signup_data)
+
+        self.enclave_data = self._get_JSON_from_signup_object(signup_data)
+        self.signup_info_handle = self.enclave_data["enclave_data"]
```

Comment:
Please add some details in a comment on what kind of data is assigned to singup_info_handle